Packs/FireMonSecurityManager/Integrations/FireMonSecurityManager/FireMonSecurityManager.py

```diff
@@ -0,0 +1,1078 @@
+                                          url_suffix=pca_url_suffix,
+                                          json_data=payload_pca,
+                                          headers=headers,
+                                          params=None,
+                                          timeout=20)
+        return api_response
+
+    def rule_rec_api(self, auth_token, payload):
+        """ Calling orchestration rulerec api by passing json data as request body, headers, params and domainId
+                which returns you list of rule recommendations for given input as response"""
+
+        parameters = {'deviceGroupId': payload["deviceGroupId"], 'addressMatchingStrategy': 'INTERSECTS',
+                      'modifyBehavior': 'MODIFY', 'strategy': None}
+        data = get_rule_rec_request_payload()
+
+        data["destinations"] = payload["destinations"]
+        data["sources"] = payload["sources"]
+        data["services"] = payload["services"]
+        data["action"] = payload["action"]
+        rule_rec_api_response = self._http_request(
+            method='POST',
+            url_suffix=RULE_REC_URL.format(payload["domainId"]),
+            json_data=data,
+            params=parameters,
+            headers={
+                'Content-Type': 'application/json',
+                'Accept': 'application/json',
+                'X-FM-Auth-Token': auth_token,
+            }
+        )
+        return rule_rec_api_response
+
+    def rule_rec_output(self, auth_token, payload):
+        """ Calling orchestration rulerec api by passing json data as request body, headers, params and domainId
+                which returns you list of rule recommendations for given input as response"""
+
+        parameters = {'deviceId': payload["deviceId"], 'addressMatchingStrategy': 'INTERSECTS',
+                      'modifyBehavior': 'MODIFY', 'strategy': None}
+        data = get_rule_rec_request_payload()
+
+        data["destinations"] = payload["destinations"]
+        data["sources"] = payload["sources"]
+        data["services"] = payload["services"]
+        data["action"] = payload["action"]
+        rule_rec_api_response = self._http_request(
+            method='POST',
+            url_suffix=RULE_REC_URL.format(payload["domainId"]),
+            json_data=data,
+            params=parameters,
+            headers={
+                'Content-Type': 'application/json',
+                'Accept': 'application/json',
+                'X-FM-Auth-Token': auth_token,
+            }
+        )
+        return rule_rec_api_response
+
+
+def test_module(client):
+    """
+     Performs basic get request to get item samples
+     """
+    response = client.authenticate_user()
+    if response.get('authorized'):
+        demisto.results("ok")
+    else:
+        demisto.results("Error in API call in FireMonSecurityManager Integrations")
+
+
+def authenticate_command(client):
+    result = client.authenticate_user().get('token')
+    return result
```

Comment:
When using `return_result` on main function we need to return from command functions a `CommandResult` object. Docs is [here](https://xsoar.pan.dev/docs/reference/api/common-server-python#commandresults)
for example: 
`    return CommandResults(
        outputs_prefix='AnomaliEnterprise.ForensicSearch',
        outputs_key_field='job_id',
        outputs=start_search_outputs,
        readable_output=tableToMarkdown(name="Forensic search started:", t=start_search_outputs, removeNull=True),
        raw_response=response
    )`

Please change it in all your command functions.

Reply:
Is this change a recommendation or required? Since implementing this particular change would touch multiple execution points including playbook, we would like to keep it to what we have. We have also seen the same usage pattern in other integrations as well, so it would be great if we can keep it the way we have and return result instead of CommandResults.

Reply:
If this is xsoar support, this change is required. I saw you didn't change it..

Reply:
resolving as outdated.